### ISSUE TITLE
Can't deserialize Float.POSITIVE_INFINITY

### ISSUE THREAD

**Issue by gunnihinn** [protostuff/protostuff#218]
Running the program

```java
public static void main(String args[]) throws Exception {
    Schema<Float> schema = RuntimeSchema.getSchema(Float.class);
    Float infinity = Float.POSITIVE_INFINITY;
    LinkedBuffer buffer = LinkedBuffer.use(new byte[1024 * 1024]);

    byte[] json = JsonIOUtil.toByteArray(infinity, schema, false, buffer);
    System.out.println(new String(json));

    Float f = schema.newMessage();
    JsonIOUtil.mergeFrom(json, f, schema, false);
    System.out.println(f);
}
```
results in the exception

```
Exception in thread "main" com.fasterxml.jackson.core.JsonParseException: 
Current token (VALUE_STRING) not numeric, can not use numeric value accessors
```

Despite what the stack trace says, the problem is in fact not in FasterXML, which does deserialize "Infinity" to `Float.POSITIVE_INFINITY`.

**Comment by dyu** [comment]
You aren't supposed to pass jdk scalar classes.  That use-case is not supported.
Always pass pojos (user-defined classes)

**Comment by gunnihinn** [comment]
I'm not sure I follow. The problem remains if I create a class that has a float field and pass that around instead.

The serialization works fine; this prints '{"value":"Infinity"}' and then throws an exception. It's the deserialization that has me puzzled.

If I'm still on the wrong track, how should I properly serialize and deserialize something that overflows a float in Java?

**Comment by kshchepanovskyi** [comment]
@dyu this is valid defect:

```java
@Test
public void testFloat_PositiveInfinity() throws Exception
{
    Bar msg = new Bar();
    msg.setSomeFloat(Float.POSITIVE_INFINITY);
    byte[] bytes = JsonIOUtil.toByteArray(msg, Bar.getSchema(), false);
    String result = new String(bytes);
    String expectedJson = "{\"someFloat\":\"Infinity\"}";
    Assert.assertEquals(expectedJson, result);
    Bar deserialized = new Bar();
    JsonIOUtil.mergeFrom(expectedJson.getBytes(), deserialized, Bar.getSchema(), false);
    Assert.assertEquals(msg, deserialized);
}
```

Test fails:

```
com.fasterxml.jackson.core.JsonParseException: Current token (VALUE_STRING) not numeric, can not use numeric value accessors
 at [Source: [B@61e717c2; line: 1, column: 15]

	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1586)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:521)
	at com.fasterxml.jackson.core.base.ParserBase._parseNumericValue(ParserBase.java:845)
	at com.fasterxml.jackson.core.base.ParserBase.getDoubleValue(ParserBase.java:759)
	at com.fasterxml.jackson.core.base.ParserBase.getFloatValue(ParserBase.java:742)
	at io.protostuff.JsonInput.readFloat(JsonInput.java:309)
	at io.protostuff.Bar.mergeFrom(Bar.java:417)
	at io.protostuff.Bar.mergeFrom(Bar.java:29)
	at io.protostuff.JsonIOUtil.mergeFrom(JsonIOUtil.java:337)
	at io.protostuff.JsonIOUtil.mergeFrom(JsonIOUtil.java:257)
	at io.protostuff.JsonIOUtil.mergeFrom(JsonIOUtil.java:239)
	at io.protostuff.JsonFloatTest.testFloat_PositiveInfinity(JsonFloatTest.java:21)
```

**Comment by laurentgo** [comment]
The following feature `JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS` can be enabled to support this use-case

**Comment by kshchepanovskyi** [comment]
Fixed in `1.5.4`.
